action.yml

```diff
@@ -288,3 +318,46 @@ runs:
       shell: bash
       run: |
         az containerapp up --name ${{ inputs.containerAppName }} --resource-group ${{ env.CA_GH_ACTION_RESOURCE_GROUP }} --image ${{ env.CA_GH_ACTION_IMAGE_TO_DEPLOY }} ${{ env.CA_GH_ACTION_TARGET_PORT_ARG }} ${{ env.CA_GH_ACTION_CONTAINER_APP_ENVIRONMENT_ARG }} ${{ env.CA_GH_ACTION_ACR_LOGIN_ARG }} ${{ env.CA_GH_ACTION_CONTAINER_APP_LOCATION_ARG }} ${{ env.CA_GH_ACTION_CONTAINER_APP_ENVIRONMENT_VARIABLES_ARG }}
+
+    - name: Mark action as 'succeeded' for telemetry
+      if: ${{ inputs.disableTelemetry == 'false' }}
+      shell: bash
+      run: |
+        CA_GH_ACTION_RESULT_ARG="--property 'result=succeeded'"
+        echo "CA_GH_ACTION_RESULT_ARG=${CA_GH_ACTION_RESULT_ARG}" >> $GITHUB_ENV
+
+    - name: End timer that's tracking action execution length, in milliseconds
+      if: ${{ always() && inputs.disableTelemetry == 'false' }}
+      shell: bash
+      run: |
+        CA_GH_ACTION_END_MILLISECONDS=$(date +%s%N | cut -b1-13)
+        CA_GH_ACTION_START_MILLISECONDS=${{ env.CA_GH_ACTION_START_MILLISECONDS }}
+        CA_GH_ACTION_LENGTH_MILLISECONDS_ARG="--processing-time '$((CA_GH_ACTION_END_MILLISECONDS-CA_GH_ACTION_START_MILLISECONDS))'"
+        echo "CA_GH_ACTION_LENGTH_MILLISECONDS_ARG=${CA_GH_ACTION_LENGTH_MILLISECONDS_ARG}" >> $GITHUB_ENV
+
+    - name: Set telemetry for previously built image scenario
+      if: ${{ inputs.disableTelemetry == 'false' && inputs.appSourcePath == '' }}
+      shell: bash
+      run: |
+        CA_GH_ACTION_SCENARIO_ARG="--property 'scenario=used-image'"
+        echo "CA_GH_ACTION_SCENARIO_ARG=${CA_GH_ACTION_SCENARIO_ARG}" >> $GITHUB_ENV
+
+    - name: Set telemetry for built Dockerfile scenario
+      if: ${{ inputs.disableTelemetry == 'false' && env.CA_GH_ACTION_DOCKERFILE_PATH != '' }}
+      shell: bash
+      run: |
+        CA_GH_ACTION_SCENARIO_ARG="--property 'scenario=used-dockerfile'"
+        echo "CA_GH_ACTION_SCENARIO_ARG=${CA_GH_ACTION_SCENARIO_ARG}" >> $GITHUB_ENV
+
+    - name: Set telemetry for no Dockerfile scenario
+      if: ${{ inputs.disableTelemetry == 'false' && inputs.appSourcePath != '' && env.CA_GH_ACTION_DOCKERFILE_PATH == '' }}
+      shell: bash
+      run: |
+        CA_GH_ACTION_SCENARIO_ARG="--property 'scenario=used-builder'"
+        echo "CA_GH_ACTION_SCENARIO_ARG=${CA_GH_ACTION_SCENARIO_ARG}" >> $GITHUB_ENV
+
+    - name: Log telemetry for action
+      if: ${{ always() && inputs.disableTelemetry == 'false' }}
+      shell: bash
+      run: |
+        docker run --rm mcr.microsoft.com/oryx/build:github-actions-debian-buster-20230206.2  /bin/bash -c "oryx telemetry --event-name 'ContainerAppsGitHubAction' ${{ env.CA_GH_ACTION_LENGTH_MILLISECONDS_ARG }} ${{ env.CA_GH_ACTION_SCENARIO_ARG }} ${{ env.CA_GH_ACTION_RESULT_ARG }}"
```

Comment:
Could you please clarify the impact of this line? Are we going to download and run the whole image here? If yes, any thoughts on the impact in terms of performance and all?

Reply:
Thanks for clarifying. Yeah that's pretty big. Let's sync during our 1:1 to see if we have any workaround.

Reply:
Update: after some testing, the time it takes to both pull the latest CLI image on MCR and run the `oryx telemetry` command is around ~4s, which is a more acceptable hit to performance.